app-games/src/main/java/cm/aptoide/pt/app_games/installer/notifications/InstallerNotificationsBuilder.kt

```diff
@@ -0,0 +1,224 @@
+        appDetails = appDetails,
+        progress = when (state) {
+          State.DOWNLOADING -> max(progress, 0)
+          State.INSTALLING,
+          State.READY_TO_INSTALL,
+          State.PENDING,
+          -> -1
+
+          else -> null
+        },
+        contentText = when (state) {
+          State.COMPLETED -> context.getString(R.string.notification_1_installed_title)
+          State.DOWNLOADING ->
+            context.getString(
+              R.string.notification_downloading_body, max(progress, 0).toString(),
+              TextFormatter.formatBytes(size)
+            )
+
+          State.INSTALLING -> context.getString(R.string.notification_installing_title)
+          else -> ""
+        }
+      )
+
+      showNotification(notificationId, notification)
+    }
+  }
+
+  suspend fun showWaitingForDownloadNotification(
+    packageName: String,
+    appDetails: AppDetails?,
+  ) {
+    val notificationId = stringToIntConverter.getStringId(packageName)
+
+    val notification = buildNotification(
+      requestCode = notificationId,
+      packageName = packageName,
+      appDetails = appDetails,
+      contentText = context.getString(R.string.notification_waiting_body),
+      progress = -1,
+    )
+
+    showNotification(notificationId, notification)
+  }
+
+  private fun cancelNotification(notificationId: Int) {
+    if (context.isAllowed(Manifest.permission.POST_NOTIFICATIONS)) {
+      NotificationManagerCompat.from(context).cancel(notificationId)
+    }
+  }
+
+  @SuppressLint("MissingPermission")
+  private fun showNotification(
+    notificationId: Int,
+    notification: Notification,
+  ) {
+    if (context.isAllowed(Manifest.permission.POST_NOTIFICATIONS)) {
+      NotificationManagerCompat.from(context).notify(notificationId, notification)
+    }
+  }
+
+  suspend fun showWaitingForWifiNotification(
+    packageName: String,
+    appDetails: AppDetails?,
+  ) {
+    val notificationId = stringToIntConverter.getStringId(packageName)
+
+    val notification = buildNotification(
+      requestCode = notificationId,
+      packageName = packageName,
+      appDetails = appDetails,
+      contentText = context.getString(R.string.notification_waiting_for_wifi_title),
+      progress = -1,
+      hasAction = true
+    )
+
+    showNotification(notificationId, notification)
+  }
+
+  private suspend fun buildNotification(
+    requestCode: Int,
+    packageName: String,
+    appDetails: AppDetails?,
+    progress: Int? = null,
+    contentText: String,
+    hasAction: Boolean = false,
+  ): Notification {
+    val deepLink =
+      buildAppViewDeepLinkUri(packageName)
+
+    val clickIntent = PendingIntent.getActivity(
+      context,
+      requestCode,
+      Intent(context, MainActivity::class.java)
+        .putDeeplink(deepLink)
+        .putNotificationSource(),
+      PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+    )
+
+    val notificationIcon = BuildConfig.FLAVOR.getNotificationIcon()
```

Comment:
This can use the notification icon directly, since for this app it does not rely on the flavor.